### DISCUSSION TITLE
Trying to figure out relationships

### DISCUSSION THREAD

**Discussion by BillThyCat** [Budibase/budibase#6451] [Help]
I have a database labeled employees.  It contains the following fields.
emp_id (employee ID)
name (full name + employee ID)
dept (craft, i.e. plumbing, electrical, etc)
foreman (auto-created by the relationship process)

I have another database named Foreman
it contains 2 fields
name (foreman's name)
emp_list (multiselect field containing multiple emp_id's)

It lets me view the list of employee's to add them to the list, so I know that part of the relationship is working.

When I want to actually use this in the app, I have an option picker to select the foreman,  I then created a multiselect field and point it at the datasource employee's, and try to filter it on the list that is stored in the foreman database and so far it just shows me all the employee's instead of an error or the correct list.

When I go into filter, there is no option for foreman in the field codes to pick from.

```
Foreman List Datasource
    Employee Datasource
        Foreman Option Picker
        Pick Employee's MultiPicker

```
It won't let me put Pick Employee indented below Foreman, it only allows them to be equally indented.

Is that why Employee Picker can't see the result of Foreman Picker?

Thanks,

**Comment by melohagan** [comment]
Hi @BillThyCat 

You need to use the **Relationship Picker** instead of the Multi-picker. https://github.com/Budibase/budibase/discussions/6274#discussioncomment-2936824

**Reply by BillThyCat** [reply]
Ok, I tried that, you can't reference what dataprovider to use within the picker.  How does it know what field and dataprovider to use for it's selection list?

Thanks,

**Reply by melohagan** [reply]
You'll need to assign a table schema to the Form that the Relationship picker is within:

![Screenshot 2022-06-28 at 09 24 31](https://user-images.githubusercontent.com/101575380/176131308-52bd1a12-c86a-4f31-99ed-89304ec79160.png)

Then you'll be able to select the relationship field from the dropdown:
![Screenshot 2022-06-28 at 09 25 27](https://user-images.githubusercontent.com/101575380/176131506-6949fcb3-7313-4ba6-ae5d-71d71aad651d.png)

**Reply by BillThyCat** [reply]
Ok, still stumped.

I need an employee list that contains the following static fields.
emp_id(numeric), emp_name(text), dept(numeric)

I need a foreman list that contains the following static fields
foreman_name(text)

I need a relationship in foreman that links to multiple employees.  so a one to many relationship.

I need to be able to use an option picker to display the foreman list
I need either a multi-option picker or relation picker that looks at the foreman selected in the option picker and only displays the employees that are linked to that foreman.

I will eventually need to be able to write the list of emp_id's from that selection to a field in my time_entry database if that makes a difference.

I tried changing the schema in the form and chose every possible option, chose the dataprovider, chose the relation field,  nothing worked.   The picker either showed nobody or everybody.  I tried to go in and create a filter on the employee dataprovider and could not find a way to only show employee's that had that particular foreman in their relation field.  The relation field did not show up as one of the choices for matching inside the filter rules.

Any chance you can build a super basic app that has two databases with those minimal fields and a basic form that contains the two pickers that allow you to select one and then only show the correct stuff on the second picker.

Export it and send me a link that I can import and just look at how you did it?  Because I have built 4 new apps from scratch (just to clear out anything that could be causing a conflict) and modified my existing app for the past several hours trying to just do that basic function and have had no luck.

I can send you an upload link if need be. Or if it's small enough can you just attach it to this message?  I'm sure I can't be the only person struggling with this, hopefully it will help others as well.

I appreciate anything you can do to help.

Thanks,

**Comment by melohagan** [comment]
Hi @BillThyCat 

This is a common use case, but also a limitation of the Internal DB. If you need to significantly rely on relationships in your app, then use a RDBMS.

This answer may also be useful to you: https://github.com/Budibase/budibase/discussions/6254#discussioncomment-3072201
